engine-types/src/account_id.rs

```diff
@@ -93,6 +92,20 @@ impl AccountId {
     }
 }
 
+impl BorshDeserialize for AccountId {
+    fn deserialize(buf: &mut &[u8]) -> io::Result<Self> {
+        let account: String = BorshDeserialize::deserialize(buf)?;
+
+        // It's for saving backward compatibility.
```

Comment:
Are there examples where we needed to deserialize an AccountId that was empty? If not then I think we should not allow empty values.

Reply:
For example, `bridge_prover_id` is empty when restoring a snapshot from `engine-tests/src/tests/res/contract.aurora.block51077328.json`. The same situation with `engine-tests/src/tests/res/aurora_state_*.json` files.

Reply:
I see. That's pretty unfortunate. Alright, we must accept empty values as valid then. Thanks for pointing out the examples.